vpr/src/route/router_lookahead.cpp

```diff
@@ -8,7 +8,7 @@
 static int get_expected_segs_to_target(int inode, int target_node, int* num_segs_ortho_dir_ptr);
 static int round_up(float x);
 
-std::unique_ptr<RouterLookahead> make_router_lookahead(e_router_lookahead router_lookahead_type) {
+static std::unique_ptr<RouterLookahead> make_router_lookahead_object(e_router_lookahead router_lookahead_type) {
```

Comment:
Can you comment on the design choice to split out the computation/loading from object construction?

Previously the constructors would initialize the lookahead objects in a valid/ready-to-use state. Now they need to be explicitly computed/loaded (the object exists in an invalid/unusable state between those times, which is potentially more bug prone).

Reply:
I wanted the same compute/write/read behavior across all object types.  Given that all nominal path code calls make_router_lookahead, I don't see a strong difference between handling this in the constructor vs what is in make_router_lookahead, and I think how ``make_router_lookahead`` is more straight forward than dealing with handling this in the constructor.